databases/week3/howework3.sql

```diff
@@ -0,0 +1,147 @@
+-- Part one Queries to write
+-- Get all meals
+
+SELECT title AS Meals
+FROM Meal;
+
+
+-- Add a new meal
+
+INSERT INTO Meal
+    ( title, description, when_create,location, max_reservations, price, created_date)
+VALUES
+    ( 'Chiken Massala', 'typical indian dishes with full of spices',
+        now(), 'Ballerupvej 43', 10, 125, '2020-09-04');
+
+-- Get a meal with any id,
+
+SELECT title, description
+FROM Meal
+WHERE id=3;
+
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE Meal
+SET description = 'Sprinkle the salt and drizzle the olive oil on top of the eggs' 
+WHERE id=3;
+
+-- Delete a meal with any id
+
+DELETE FROM Meal
+WHERE id = 7;
+
+-- Part 2
+-- Queries to write
+
+-- Get all reservations
+
+SELECT *
+FROM Reservation;
+
+-- Add a new reservation
+
+INSERT INTO Reservation
+    (number_of_guests, meal_id, created_date, contact_name, contact_email, contact_phonenumber)
+values
+    ( 2, 1, '2020-10-06', 'Pavel Brushneen', 'pbrushneen6@techcrunch.com', '316-170-3640');
+
+-- Get a reservation with any id
+
+SELECT *
+FROM Reservation
+WHERE id = 8;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE Reservation
+SET contact_phonenumber ='891-952-6749',contact_name = 'John Smith'
+WHERE id=8;
+
+-- Delete a reservation with any id, fx 1
+DELETE FROM Reservation
+WHERE id = 7;
+
+-- Part3 Queries to write in Reviews
+
+-- Get all reviews
+
+SELECT *
+FROM Review;
+
+-- Add a new review
+INSERT INTO Review
+    ( title,description, meal_id, stars)
+VALUES
+    ('good', 'taste like always', 2, 4);
+
+-- Get a review with any id, fx 1
+SELECT *
+FROM Review
+WHERE id = 3;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE Review
+SET title='low',stars = 2
+WHERE id=5;
+
+-- Delete a review with any id, fx 1
+
+DELETE FROM Review
+WHERE id = 7;
+
+-- Get meals that has a price smaller than a specific price fx 90
+
+SELECT title
+FROM Meal
+WHERE price < 90;
+
+-- Get meals that still has available reservations
+SELECT Meal.title
+FROM Meal
+    JOIN Reservation
+    ON Meal.id = Reservation.meal_id
+WHERE Reservation.created_date > CURRENT_TIMESTAMP;
```

Comment:
I think what is being referred to here in `Get meals that still has available reservations` is to find meals that have less than `max_reservations` number of reservations. How would that be done?

Reply:
@mclemme I got the concept now! wrote this query to get the total number_of_guests (orders) from Reservation table! but still don't know how to reduce them from max_reservation in Meal table!!  any help or hint? I did searched but reluctant!
this is the query to get total of meal reservation:
`SELECT meal_id, 
SUM(number_of_guests)
FROM Reservation
GROUP BY meal_id
Order by meal_id`

Reply:
@arya56 If we assume that **Meal.max_reservations** is the field representing the amount of seats left at a given meal.

So for the sake of example, let's say we have a meal with the ID **42** and  value **10** in the **max_reservations** field. This would mean that this particular meal could seat 10 people in total.

Then furthermore let's say insert two reservations in the **Reservation** table like this:

`INSERT INTO `Reservation` VALUES (NULL, 4, 42, '2020-10-30', 555-1234', 'Donald Duck', 'donald@duck.example')`
and
`INSERT INTO `Reservation` VALUES (NULL, 4, 42, '2020-10-29', 555-1235', 'Mickey Mouse', 'mickey@mouse.example')`

Then we'd have `10 - 4 - 4 = 2` seats left at that meal, right?

So you need to sum up all the **Reservation.number_of_guests** grouped by which meal the reservations belong to and find out if the sum of guests are lower than the **Meal.max_reservations**. There are many ways to solve this, but I'd suggest you look at the following SQL keywords:

- `JOIN`
- `GROUP BY`
- `HAVING`

Reply:
Thanks @mclemme for this great comment.